ppafm/io.py

```diff
@@ -471,224 +471,304 @@ def saveXSF(fname, data, lvec=None, dd=None, head=XSF_HEAD_DEFAULT, verbose=1 ):
+        lvec: np.array of shape (4,3) with lattice vector of the volumetric data
+        data_format: string "xsf" or "npy"
+        head: string header of the XSF file
+        atomic_info: tuple of shape (2) with 2 np.arrays - one is np.array([e,x,y,z]) with atoms positions and the second one is np.array(lvec) of shape (4,3) with saved information about lattice vector.
+    '''
+    if (data_format=="xsf"):
+        saveVecFieldXsf(fname, data, lvec, head = head )
+    elif (data_format=="npy"):
+        atomic_info = atomic_info if atomic_info is not None else (np.zeros((4,1)),lvec)
+        saveVecFieldNpy(fname, data, lvec, atomic_info )
+    else:
+        print("I cannot save this format!")
 
 def load_vec_field(fname, data_format="xsf"):
-	'''
-	Loading Vector fields into xsf, or npy
-	'''
-	if (data_format=="xsf"):
-		data, lvec, ndim, head = loadVecFieldXsf(fname)
-	elif (data_format=="npy"):
-		data, lvec = loadVecFieldNpy(fname)
-		ndim = np.delete(data.shape,3)
-	else:
-		print("I cannot load this format!")
-	return data, lvec, ndim
+    '''
+    Loading Vector fields from xsf, or npy
+
+    Arguments:
+        fname: str. name of the npz file.
+        data_fromat: str "xsf" or "npy"
+
+    Returns:
+        data: np.array of shape(nz, ny, nx, 3) with volumetric (vector data) we want to load.
+        lvec: np.array of shape(4,3) with lattice vector of the volumetric data.
+        atomic_info_or_heaad: tuple or string. If tupple then shape (2) with 2 np.arrays,
+             one is np.array([e,x,y,z]) with atoms positions and the second one is np.array(lvec0) of shape (4,3) with saved information about lattice vector.
+             if string, the same information is basically stored as the header of xsf
+    '''
+    atomic_info_or_head = None
+    if (data_format=="xsf"):
+        data, lvec, ndim, atomic_info_or_head = loadVecFieldXsf(fname)
+    elif (data_format=="npy"):
+        data, lvec, atomic_info_or_head = loadVecFieldNpy(fname)
+        ndim = data.shape
+    else:
+        print("I cannot load this format!")
+    return data.copy(), lvec, ndim, atomic_info_or_head;
 
 
 # =============== Scalar Fields
 
-def save_scal_field(fname, data, lvec, data_format="xsf", head = XSF_HEAD_DEFAULT ):
-	'''
-	Saving scalar fields into xsf, or npy
-	'''
-	if (data_format=="xsf"):
-		saveXSF(fname+".xsf", data, lvec, head = head)
-	elif (data_format=="npy"):
-		saveNpy(fname, data, lvec, head = head)
-	else:
-		print("I cannot save this format!")
+def save_scal_field(fname, data, lvec, data_format="xsf", head = XSF_HEAD_DEFAULT , atomic_info = None):
+    '''
+    Saving scalar fields into xsf, or npy
+
+    Arguments:
+        fname: str. name of the npz file.
+        data: np.array of shape(nz, ny, nx, 3) with volumetric (scalar data) we want to save/
+        data_fromat: str "xsf" or "npy"/
+        lvec: np.array of shape(4,3) with lattice vector of the volumetric data.
```

Comment:
Probably better to put the argument descriptions in the same order as the actual arguments, so `lvec` here would be above `data_format`.

Missing description of the `head` argument.